### PR TITLE
Extend Video Mime type support for AMP Story

### PR DESCRIPTION
**Pull request by thrijith**
- Extend support for video mime type in AMP Story Background Media and Video Block

Depends on https://github.com/ampproject/amp-wp/pull/3171

### CODE REVIEW

File: admin/rt-retranscode-admin.php [hunk 1]
```diff
@@ -633,6 +635,36 @@ public function transcoded_thumbnails_added( $media_id = '' ) {
...
+		$thumbnail_array = get_post_meta( $media_id, '_rt_media_thumbnails', true );
+
+		if ( is_array( $thumbnail_array ) ) {
+			$uploads   = wp_upload_dir();
+			$thumbnail = $thumbnail_array[0];
+
+			if ( 0 === strpos( $thumbnail, $uploads['baseurl'] ) ) {
+				$thumbnail_src = $thumbnail;
+			} else {
+				$thumbnail_src = $uploads['basedir'] . '/' . $thumbnail;
```
Comment (dharmin) [inline admin/rt-retranscode-admin.php]:
```suggestion
				$tumbnail_src = trailingslashit( $uploads['basedir']  ) . $thumbnail;
```

File: admin/rt-retranscode-admin.php [hunk 2]
```diff
@@ -655,6 +687,92 @@ public function rtt_handle_callback_finished( $attachment_id = '', $job_id = ''
...
+				if ( ! empty( $transcoded_url ) && isset( $transcoded_url['mp4'] ) ) {
+					// Get transcoded video path.
+					$transcoded_url = empty( $transcoded_url['mp4'][0] ) ? '' : $transcoded_url['mp4'][0];
+					$uploads        = wp_get_upload_dir();
+
+					// Get URL for the transcoded video.
+					if ( 0 === strpos( $transcoded_url, $uploads['baseurl'] ) ) {
+						$final_file_url = $transcoded_url;
+					} else {
+						$final_file_url = $uploads['baseurl'] . '/' . $transcoded_url;
```
Comment (dharmin) [inline admin/rt-retranscode-admin.php]:
```suggestion
						$final_file_url = trailingslashit( $uploads['baseurl'] ) . $transcoded_url;
```

File: admin/rt-retranscode-admin.php [hunk 3]
```diff
@@ -633,6 +635,36 @@ public function transcoded_thumbnails_added( $media_id = '' ) {
...
+				'post_mime_type' => $file_type['type'],
+				'post_title'     => preg_replace( '/\.[^.]+$/', '', basename( $thumbnail_src ) ),
+				'post_content'   => '',
+				'post_status'    => 'inherit'
+			);
+
+			$attachment_id = wp_insert_attachment( $attachment, $thumbnail_src, $media_id );
+
+			require_once( ABSPATH . 'wp-admin/includes/image.php' );
+			$attach_data = wp_generate_attachment_metadata( $attachment_id, $thumbnail_src );
```
Comment (dharmin) [inline admin/rt-retranscode-admin.php]: Should check is_wp_error for $attachment_id.

**Comment by pradeep910** [comment]
@dharmin @pooja-muchandikar is this well tested?

**Pull request merged**